variables.tf

```diff
@@ -0,0 +1,40 @@
+variable "backend_config" {
+  description = "Object of S3 backend config"
+  type = object({
+    bucket        = string
+    force_destroy = optional(bool)
+    versioning    = optional(string)
+    public_access_block = optional(object({
+      block_public_acls       = bool
+      block_public_policy     = bool
+      ignore_public_acls      = bool
+      restrict_public_buckets = bool
+    }))
+    server_side_encryption_configuration = optional(object({
+      bucket_key_enabled = bool
+      sse_algorithm      = string
+      kms_master_key_id  = string
+    }))
+    dynamodb_table = optional(object({
+      name                        = string
+      deletion_protection_enabled = bool
+      billing_mode                = string
+    }))
+  })
+  default = {
+    bucket        = ""
+    force_destroy = "true"
+    versioning    = "Enabled"
+    public_access_block = {
+      block_public_acls       = true
+      block_public_policy     = true
+      ignore_public_acls      = true
+      restrict_public_buckets = true
+    }
+    server_side_encryption_configuration = {
+      bucket_key_enabled = true
+      sse_algorithm      = "aws:kms"
+      kms_master_key_id  = null
+    }
+  }
```

Comment:
Delete this default value. The user should be required to provide this input variable.

```suggestion
```